tests/build_/test_arrangements.py

```diff
@@ -0,0 +1,248 @@
+        'user': 'john-foo',
+        'component': TEST_COMPONENT,
+        'base_image': 'fedora:latest',
+        'name_label': 'fedora/resultingimage',
+        'registry_uri': 'registry.example.com',
+        'pulp_registry': 'pulp-registry.example.com',
+        'pulp_secret': 'pulp-secret',
+        'source_registry_uri': 'registry.example.com',
+        'openshift_uri': 'http://openshift/',
+        'builder_openshift_url': 'http://openshift/',
+        'koji_target': 'koji-target',
+        'kojiroot': 'http://root/',
+        'sources_command': 'make',
+        'vendor': 'Foo Vendor',
+        'authoritative_registry': 'registry.example.com',
+        'distribution_scope': 'authoritative-source-only',
+        'registry_api_versions': ['v1', 'v2'],
+        'smtp_host': 'smtp.example.com',
+        'smtp_from': 'user@example.com',
+        'proxy': 'http://proxy.example.com',
+    }
+
+    ORCHESTRATOR_ADD_PARAMS = {
+        'platforms': ['x86_64'],
+    }
+
+    WORKER_ADD_PARAMS = {
+        'platform': 'x86_64',
+        'release': 1,
+    }
+
+    def mock_env(self, base_image=None):
+        class MockParser(object):
+            labels = {
+                'name': 'fedora23/something',
+                'com.redhat.component': TEST_COMPONENT,
+                'version': TEST_VERSION,
+            }
+            baseimage = base_image or 'fedora23/python'
+
+        (flexmock(utils)
+            .should_receive('get_df_parser')
+            .with_args(TEST_GIT_URI, TEST_GIT_REF, git_branch=TEST_GIT_BRANCH)
+            .and_return(MockParser()))
+
+        # Trick create_orchestrator_build into return the *request* JSON
+        flexmock(OSBS, _create_build_config_and_build=request_as_response)
+        flexmock(OSBS, _create_scratch_build=request_as_response)
+
+    def get_orchestrator_build_request(self, osbs,  # noqa
+                                       additional_params=None):
+        self.mock_env(base_image=additional_params.get('base_image'))
+        params = self.COMMON_PARAMS.copy()
+        params.update(self.ORCHESTRATOR_ADD_PARAMS)
+        params.update(additional_params or {})
+        params['arrangement_version'] = self.ARRANGEMENT_VERSION
+        return params, osbs.create_orchestrator_build(**params).json
+
+    def get_worker_build_request(self, osbs, additional_params=None):  # noqa
+        self.mock_env(base_image=additional_params.get('base_image'))
+        params = self.COMMON_PARAMS.copy()
+        params.update(self.WORKER_ADD_PARAMS)
+        params.update(additional_params or {})
+        params['arrangement_version'] = self.ARRANGEMENT_VERSION
+        return params, osbs.create_worker_build(**params).json
+
+    def assert_plugin_not_present(self, build_json, phase, name):
+        plugins = get_plugins_from_build_json(build_json)
+        with pytest.raises(NoSuchPluginException):
+            get_plugin(plugins, phase, name)
+
+    @pytest.mark.parametrize('base_image', [  # noqa
```

Comment:
Why is `#noqa` used here? Usually, this is done because flake8 doesn't know about py.test fixtures. But looking at the params in your test method, all of them appear to be used in the method body.